src/emailjs-imap-client.js

```diff
@@ -250,25 +251,26 @@
         }
 
         this.logger.debug('Opening', path, '...');
-        return this.exec(query, ['EXISTS', 'FLAGS', 'OK'], {
-            ctx: options.ctx
-        }).then((response) => {
-            this._changeState(this.STATE_SELECTED);
-
-            if (this._selectedMailbox && this._selectedMailbox !== path) {
-                this.onclosemailbox && this.onclosemailbox(this._selectedMailbox);
-            }
-
-            this._selectedMailbox = path;
+        return new Promise((resolve, reject) =>
+            this._selectMailboxLock = this._selectMailboxLock.then(() =>
+                this.exec(query, ['EXISTS', 'FLAGS', 'OK'], {
```

Comment:
returning this then-able rather than nesting once-further will make it easier to read :)